src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1279,7 +1280,42 @@ private void Abort(Exception abortException)
             return LifetimeExpired(nowTicks, connectionLifetime);
         }
 
-        private void AbortStreams(int lastValidStream, Exception abortException)
+        private void AbortStreams(Exception abortException)
+        {
+            lock (SyncObject)
+            {
+                if (NetEventSource.IsEnabled) Trace($"{nameof(abortException)}={abortException}, {nameof(_lastValidStreamId)}={_lastValidStreamId}");
+
+                bool hasAnyActiveStream = false;
+                foreach (KeyValuePair<int, Http2Stream> kvp in _httpStreams)
+                {
+                    int streamId = kvp.Key;
+                    Debug.Assert(streamId == kvp.Value.StreamId);
+
+                    kvp.Value.OnAbort(abortException);
+                    _httpStreams.Remove(streamId);
+                }
+
+                if (!hasAnyActiveStream)
+                {
+                    _disposed = true;
+                }
+
+                CheckForShutdown();
+            }
+
+            if (!_disposed)
```

Comment:
Per above, this test will never succeed, right?

Reply:
good catch, I suspect it might have caused some flakiness in the test which I'm seeing (although not sure)

Reply:
actually lack of that was causing hangs which have incidentally passing the test run. I'll do an extra PR to make sure we fail when we hang (will do it separately since there are couple of tests which use similar structure)